agents/plugins/yum

```diff
@@ -1,125 +1,116 @@
 #!/bin/bash
 
+if [ -z $MK_VARDIR ]; then
+    echo "Unable to load ENV variables"
+    exit 2
+fi
+
 BOOT_REQUIRED=no
 UPDATES=0
 SECURITY_UPDATES=0
+CACHE_RESULT_CHECK=$MK_VARDIR/cache/yum_result.cache
+CACHE_YUM_UPDATE=$MK_VARDIR/cache/yum_update.cache
 
-echo "<<<yum>>>"
-
-# get curent yum state - use cache directory contents as fingerprint 
+# get current yum state - use cache directory contents as fingerprint
 YUM_CURRENT="$(ls -lR /var/cache/yum/ || ls -lR /var/cache/dnf/)"
 
 # check if cached listing of /var/cache/yum already exists - create empty one otherwise
-if [ ! -e $MK_VARDIR/cache/yum_state.cache ]
+if [ ! -e $CACHE_YUM_UPDATE ]
 then
-    touch $MK_VARDIR/cache/yum_state.cache
-elif [ ! -f $MK_VARDIR/cache/yum_state.cache ] || [ -L $MK_VARDIR/cache/yum_state.cache ]
+    touch $CACHE_YUM_UPDATE
+elif [ ! -f $CACHE_YUM_UPDATE ] || [ -L $CACHE_YUM_UPDATE ]
 then
     # something is wrong here...
     echo "invalid cache file"
-    exit
+    exit 2
+else
+    # get cached information
+    YUM_CACHED=$(cat "$CACHE_YUM_UPDATE")
 fi
 
 # check if cached check result already exists and is nothing but a file
-# newer CMK (1.6?) shall be checked to be backward compatible
-YUM_CACHE_FILE="$MK_VARDIR/cache/$(ls -t1 $MK_VARDIR/cache | grep yum.cache)" 
-# cut off the .new suffix as it is vanished later anyway
-YUM_CACHE_FILE="${YUM_CACHE_FILE%%.new}"
-
-if [ ! -e "$YUM_CACHE_FILE" && ! -f "$YUM_CACHE_FILE" ]
+if [ ! -e $CACHE_RESULT_CHECK  ]
 then
-    # fallback without plugin
-    if [ ! -e $MK_VARDIR/cache/yum_check.cache ]
-    then
-        touch $MK_VARDIR/cache/yum_check.cache
-	YUM_CACHE_FILE=$MK_VARDIR/cache/yum_check.cache
-    elif [ ! -f $MK_VARDIR/cache/yum_check.cache ] || [ -L $MK_VARDIR/cache/yum_check.cache ]
-    then
-        # something is wrong here...
-        echo "invalid cache file"
-        exit
-    fi
+    touch $CACHE_RESULT_CHECK
+elif [ ! -f $CACHE_RESULT_CHECK ] || [ -L $CACHE_RESULT_CHECK ]
+then
+    # something is wrong here...
+    echo "invalid cache file"
+    exit 2
 fi
 
 # check if system has rebooted - if so, remove cached check file to avoid wrong "reboot required"-state
 RUNNING_SECS=$(cat /proc/uptime | cut -d" " -f1 | cut -d"." -f1)
 
 # assume 120 seconds for reboot being enough
-if (( $RUNNING_SECS < 120 ))
+if (( RUNNING_SECS < 120 ))
```

Comment:
Shouldn't be the $ here?

Reply:
It isn't necessaire and could be lead to un unwonted expansion

Reply:
Ah OK, I just thought it was a typo.